### PR TITLE
Connect adjacent insNext and insPrev when running purge

### PR DESCRIPTION
**Pull request by hackerwins**
**What this PR does / why we need it**:

Connect adjacent insNext and insPrev when running purge.

In #104, GC was also introduced for text nodes marked with tombstones. However, in `release()`, the link between adjacent insPrev and insNext is missing, so this commit fixes it.

block-wise-rga

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: pkg/document/json/rga_tree_split.go [hunk 1]
```diff
@@ -580,22 +580,19 @@ func (s *RGATreeSplit) cleanupRemovedNodes(ticket *time.Ticket) int {
...
-	node.next, node.prev = nil, nil
+	node.prev, node.next = nil, nil
 
 	if node.insPrev != nil {
-		node.insPrev.insNext, node.insPrev = nil, nil
+		node.insPrev.insNext = node.insNext
 	}
 	if node.insNext != nil {
-		node.insNext.insPrev, node.insNext = nil, nil
+		node.insNext.insPrev = node.insPrev
```
Comment (dc7303) [inline pkg/document/json/rga_tree_split.go:595]: I implemented it in a way that breaks the relationship altogether because the split link represents the relationship that occurs when the node is split. I'm curious about a possible problem when not connecting `insPrev` and `insNext`.

File: pkg/document/json/root_test.go [hunk 2]
```diff
@@ -26,33 +26,53 @@ import (
...
-func registerTextElement(fromPos, toPos *json.RGATreeSplitNodePos, root *json.Root, text json.TextElement) {
+func registerRemovedNodeTextElement(fromPos, toPos *json.RGATreeSplitNodePos, root *json.Root, text json.TextElement) {
 	if fromPos.Compare(toPos) != 0 {
 		root.RegisterRemovedNodeTextElement(text)
 	}
 }
 
 func TestRoot(t *testing.T) {
-	t.Run("garbage collect for text test", func(t *testing.T) {
+	t.Run("garbage collection for array test", func(t *testing.T) {
```
Comment (dc7303) [inline pkg/document/json/root_test.go:36]: This test seems to be out of scope for this PR. Any reason added in this PR?

**Review by dc7303 — COMMENTED** [review]
Thank you for your great work! 👍  I had a question, so I left a comment. Please check.

File: pkg/document/json/root_test.go (hunk 2 above)
Comment (hackerwins) [reply pkg/document/json/root_test.go:36]: Disconnecting `node.prev` and `node.next` is missing when purging a node from RGATreeList. As I added related tests in this PR, I changed the names of other adjacent tests as well.

File: pkg/document/json/rga_tree_split.go (hunk 1 above)
Comment (hackerwins) [reply pkg/document/json/rga_tree_split.go:595]: `insPrev` and `insNext` are used to remember other nodes connected to the insert. For example, when `abc` is divided into `a`, `b`, `c`, it looks like this: `[abc]` divided to `[a]<->[b]<->[c]` Even if we completely delete `b`, I think it should keep the relationship of `[a]` <-> `[c]`.

`insPrev` is used by `findFloorNodePreferToLeft`. I think that it will not work properly when looking for the left node with priority from the given location.

https://github.com/yorkie-team/yorkie/blob/master/pkg/document/json/rga_tree_split.go#L342

Comment (dc7303) [reply pkg/document/json/rga_tree_split.go:595]: Oh! Thank you for the clear explanation :D

**Review by dc7303 — APPROVED** [review]
LGTM 👍  Thanks to you, I learned a lot!
